android/app/src/main/java/org/fossasia/openevent/dbutils/DatabaseOperations.java

```diff
@@ -693,6 +705,9 @@ public Microlocation getLocationByName(String speakerName, SQLiteDatabase mDb) {
         if (cursor.getCount() > 0) {
             while (!sessionCursor.isAfterLast()) {
                 try {
+                    int id = cursor.getInt(cursor.getColumnIndex(DbContract.Sessions.MICROLOCATION));
```

Comment:
On opening a single item in Microlocation fragment, I get a NPE.
Reason for it is that the `id` is null here because you have already closed the cursor above.
On commenting the `cursor.close()` command and running the app again, I get a Force close with the following message: 
```
Caused by: java.lang.IllegalStateException: Couldn't read row 0, col -1 from CursorWindow.  Make sure the Cursor is initialized correctly before accessing data from it.
                      at android.database.CursorWindow.nativeGetLong(Native Method)
                      at android.database.CursorWindow.getLong(CursorWindow.java:511)
                      at android.database.CursorWindow.getInt(CursorWindow.java:578)
                      at android.database.AbstractWindowedCursor.getInt(AbstractWindowedCursor.java:69)
                      at org.fossasia.openevent.dbutils.DatabaseOperations.getSessionbyLocationname(DatabaseOperations.java:708)
                      at org.fossasia.openevent.dbutils.DbSingleton.getSessionbyLocationName(DbSingleton.java:145)
                      at org.fossasia.openevent.activities.LocationActivity.onCreate(LocationActivity.java:64)
```